osquery/tables/system/linux/process_open_pipes.cpp

```diff
@@ -0,0 +1,193 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed in accordance with the terms specified in
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+#include <iostream>
+#include <osquery/core.h>
+#include <osquery/filesystem/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <regex>
+
+namespace osquery {
+namespace tables {
+
+struct pipe_info {
+  pid_t pid;
+  int fd;
+  std::string mode;
+  ino_t inode;
+  std::string type;
+  pipe_info(pid_t pid, int fd, std::string mode, ino_t inode, std::string type)
+      : pid(pid), fd(fd), mode(mode), inode(inode), type(type) {}
+};
+
+using PidToPipesMap = std::map<pid_t, std::vector<pipe_info*>>;
+using InodeToPipesMap = std::map<ino_t, std::vector<pipe_info*>>;
+
+bool isSamePipe(const pipe_info* p1, const pipe_info* p2) {
```

Comment:
Similar to @theopolis' suggestions: these should probably take `const pipe_info& p` references instead of raw pointers.